client/checkout/platform-checkout/express-button/utils.js

```diff
@@ -0,0 +1,16 @@
+/* global wcpayWooPayExpressParams, wc */
+
+/**
+ * Retrieves a configuration value.
+ *
+ * @param {string} name The name of the config parameter.
+ * @return {*}         The value of the parameter of null.
+ */
+export const getWooPayExpressData = ( name ) => {
+	// Config for the Edit Order screen.
+	const config =
+		wcpayWooPayExpressParams ??
+		wc.wcSettings.getSetting( 'woocommerce_payments_data' );
```

Comment:
I have a quick question here. From the description, I assume this function is inspired from [this](https://github.com/Automattic/woocommerce-payments/blob/add/1401-woopay-express-checkout-functionality/client/utils/checkout.js#L9-L17) or [this](https://github.com/Automattic/woocommerce-payments/blob/add/1401-woopay-express-checkout-functionality/client/utils/order.js#L9-L16).

My understanding is the `wcpay_config` or `wcpay_order_config` is present for classic shortcode cart/checkout and the `wc.wcSettings.getSetting( 'woocommerce_payments_data' )` is present on the Blocks cart/checkout.

In our case shouldn't this `wcpayWooPayExpressParams` be present on both shortcode/blocks pages? Do we need this fallback? Or does this fallback contain the correct data? :thinking:

Reply:
I removed the fallback altogether from this function, since as @Mayisha mentioned that variable should be available in any scenario where the button is being loaded with functionality anyway. However, I did have a hard time with getting these JS params passed through in a way that made sense to me, so I'm very much open to feedback on better approaches for handling this rather than the myriad of fallbacks and different utility functions, even if it gets handled in a separate issue?